Makefile

```diff
@@ -18,7 +19,18 @@ clean: ## archive and delete most git-ignored files
 	tar xf $(PRIVATE_FILES)
 	rm $(PRIVATE_FILES)
 
-docs: ## build the developer documentation for this repository
+SWAGGER = docs/swagger.yaml
+
+docs: api-docs guides ## build all the developer documentation for this repository
+
+swagger: ## generate the swagger.yaml file
+	DJANGO_SETTINGS_MODULE=docs.docs_settings python manage.py lms generate_swagger --generator-class=openedx.core.openapi.ApiSchemaGenerator -o $(SWAGGER)
```

Comment:
Cross-posting [this reply from the forum](https://discuss.openedx.org/t/one-swagger-yaml-or-many/352/6) -- is it possible to use a `swagger.yaml` file for each djangoapp, to make this more plugin-friendly?  Are there any disadvantages to this?

Reply:
The swagger/OpenAPI culture seems to be that a swagger.yaml file corresponds to a server.  This pull request follows that model, and produces one file that includes all the endpoints, regardless of whether they are implemented in this repo, or pulled in through a dependency.

I think pluggable Django apps participate in this just fine.  I didn't realize it before, but the bookmark app that I've been experimenting with is a plugged app (it's added to INSTALLED_APPS with the plugin_apps.get_apps line).

Reply:
I think what you are saying is that the optionality of a plugin means that it is hard to tell any given client what API is available to them? Or to publish the docs for a plugin before it has been installed in any given edx-platform yet?  We have that problem with the feature toggles in edx-platform also, which is why docs/doc-settings forces all toggles to True, so that everything will be in the static docs.

One of the things we have to work out is how documentation should indicate that kind of optionality. There might be more tooling that we need for it.